### DISCUSSION TITLE
how to add tag in topology service?

### DISCUSSION THREAD

**Discussion by Mushano** [apache/skywalking#10615] [Q&A] [answered]
I want to add a custom tag for each service so that I can group based on tag

I'm using version 9.3, but I don't see a button to set tag in current topology

![image](https://user-images.githubusercontent.com/38551745/228713662-bf35918e-9971-4c81-bca8-29ec243f9ad2.png)

**Comment by wu-sheng** [comment, marked as answer]
We don't have the tag for service or topology.

**Reply by Mushano** [reply]
Thank you for your answer! May I ask, do these metrics associated with topology nodes support customization? For example, I want to add a metric called app_id to show relevant information.
![image](https://user-images.githubusercontent.com/38551745/228744437-da3635d5-a6d9-4fca-af74-fba8936d46f2.png)

**Reply by wu-sheng** [reply]
Yes. Learn OAL and MAL to build your own metrics. This page would automatically show those when the OAP has those metrics.
